src/main/java/gyro/google/compute/RegionDiskResource.java

```diff
@@ -0,0 +1,268 @@
+        List<ValidationError> errors = new ArrayList<>();
+
+        if (getReplicaZones().size() != 2) {
+            errors.add(new ValidationError(
+                this,
+                "replica-zones",
+                "Disk requires exactly two replica zones."));
+        }
+
+        return errors;
+    }
+
+    @Override
+    public boolean refresh() {
+        Compute client = createComputeClient();
+
+        try {
+            Disk disk = client.regionDisks().get(getProjectId(), getRegion(), getName()).execute();
+            copyFrom(disk);
+
+            return true;
+        } catch (GoogleJsonResponseException je) {
+            if (je.getDetails().getMessage().matches("The resource (.*) was not found")) {
+                return false;
+            } else {
+                throw new GyroException(je.getDetails().getMessage());
+            }
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public void delete(GyroUI ui, State state) {
+        Compute compute = createComputeClient();
+
+        try {
+            compute.regionDisks().delete(getProjectId(), getRegion(), getName()).execute();
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    @Override
+    public void update(GyroUI ui, State state, Resource current, Set<String> changedFieldNames) {
+        Compute client = createComputeClient();
+
+        if (changedFieldNames.contains("size-gb")) {
+            saveSizeGb(client, (RegionDiskResource) current);
+        }
+
+        if (changedFieldNames.contains("labels")) {
+            saveLabels(client);
+        }
+
+        if (changedFieldNames.contains("resource-policies")) {
+            saveResourcePolicies(client, (RegionDiskResource) current);
+        }
+
+        refresh();
+    }
+
+    private void saveSizeGb(Compute client, RegionDiskResource oldRegionDiskResource) {
+        if (getSizeGb() < oldRegionDiskResource.getSizeGb()) {
+            throw new GyroException("Size of the disk cannot be decreased once set.");
+        }
+
+        try {
+            RegionDisksResizeRequest resizeRequest = new RegionDisksResizeRequest();
+            resizeRequest.setSizeGb(getSizeGb());
+            client.regionDisks().resize(getProjectId(), getRegion(), getName(), resizeRequest).execute();
+        } catch (IOException ex) {
+            throw new GyroException(ex.getMessage(), ex.getCause());
+        }
+    }
+
+    private void saveLabels(Compute client) {
+        try {
+            RegionSetLabelsRequest labelsRequest = new RegionSetLabelsRequest();
+            labelsRequest.setLabels(getLabels());
+            labelsRequest.setLabelFingerprint(getLabelFingerprint());
+            client.regionDisks().setLabels(getProjectId(), getRegion(), getName(), labelsRequest).execute();
+        } catch (IOException ex) {
```

Comment:
Should catch the `GoogleJsonResponseException`, which has better details of the error message.
```suggestion
        } catch (GoogleJsonResponseException ex) {
```